src/phase0/state_transition/mod.rs

```diff
@@ -485,3 +491,358 @@ pub fn compute_fork_data_root(
+    let epoch = compute_epoch_at_slot(slot, context);
+    let committees_per_slot = get_committee_count_per_slot(state, epoch);
+    let indices = get_active_validator_indices(&state, epoch);
+    let seed = get_seed(&state, epoch, DomainType::BeaconAttester);
+    let index = (slot % SLOTS_PER_EPOCH) * committees_per_slot + index as u64;
+    let count = committees_per_slot * SLOTS_PER_EPOCH;
+    compute_committee(
+        indices.as_slice(),
+        seed,
+        index as usize,
+        count as usize,
+        context,
+    )
+}
+
+pub fn get_beacon_proposer_index<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const PENDING_ATTESTATIONS_BOUND: usize,
+    const MAX_COMMITTEES_PER_SLOT: u64,
+    const SLOTS_PER_EPOCH: u64,
+    const TARGET_COMMITTEE_SIZE: u64,
+>(
+    state: &BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        PENDING_ATTESTATIONS_BOUND,
+    >,
+    context: &Context,
+) -> Result<ValidatorIndex, Error> {
+    let epoch = get_current_epoch(state, context);
+    let mut input = [0u8; 40];
+    input[..32].copy_from_slice(get_seed(state, epoch, DomainType::BeaconProposer).as_ref());
+    input[32..40].copy_from_slice(&state.slot.to_le_bytes());
+    let seed = hash(input);
+    let indices = get_active_validator_indices(state, epoch);
+    compute_proposer_index(state, indices.as_slice(), &seed, context)
+}
+
+pub fn get_total_balance<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const PENDING_ATTESTATIONS_BOUND: usize,
+    const MAX_COMMITTEES_PER_SLOT: u64,
+    const SLOTS_PER_EPOCH: u64,
+    const TARGET_COMMITTEE_SIZE: u64,
+    const EFFECTIVE_BALANCE_INCREMENT: u64,
+>(
+    state: &BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        PENDING_ATTESTATIONS_BOUND,
+    >,
+    indices: Vec<ValidatorIndex>,
+    context: &Context,
+) -> Result<Gwei, Error> {
+    let total_balance = indices
+        .into_iter()
+        .try_fold(0usize, |acc, i| {
+            acc.checked_add(state.validators[i].effective_balance as usize)
+        })
```

Comment:
is there a reason we can't stick with `u64` arithmetic here?

`try_fold` with `0u64` starting accumulator?

Reply:
sure